lib/3scale/api/client.rb

```diff
@@ -351,6 +396,12 @@ def extract(collection: nil, entity:, from:)
           raise "unknown #{from}"
         end
       end
+
+      private
+
+      def raise_when_empty_param(param, param_name)
+        raise "empty #{param_name}" if param.nil? || param.to_s.empty?
```

Comment:
I don't really see the value in raising when the param is empty. It doubles the complexity for no apparent benefit because I'm not aware we had any issue regarding empty IDs. This looks like you are trying to reproduce a type system in a dynamic language.

Also, when raising "validation" errors, it makes sense to always use tagged exceptions and not just RuntimeError.

Reply:
I will change error type.

But extra checking is always helpful and avoids cases when 

>DELETE /admin/api/services/#{id}

and `id` is empty could be problematic.

Reply:
I don't really see it as problematic, because the API is designed properly and will handle that just fine. We haven't had any real issues, so I don't see the value of solving an imaginary problem. But it is fine, we just disagree.